### PR TITLE
add avro docs and enrich examples

### LINKED ISSUE #321: Document source format (Protobuf & JSON & Avro)
### Related code PR

_No response_

### Which part(s) of the docs might be affected or should be updated? And how?

https://www.risingwave.dev/docs/latest/sql-create-source/
Need to call out that stream formats are specified with `ROW FORMAT ...`. 

### Reference

_No response_

### PR DESCRIPTION
**Pull request by hengm3467**
## Info
- **Description**: 
Add docs for Avro and examples for different formats under each connector

- **Related code PR**: 
Resolves #321 

- **Related doc issue**: 
Resolves [ Provide the link to the doc issue here. ]

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>image</details>

## Published doc pages
  [ After merging this PR, edit the description to include the links to the updated doc pages here. For example, https://www.risingwave.dev/docs/latest/intro/. ]

### CODE REVIEW

File: docs/sql/create-source/create-source-kafka-redpanda.md [hunk 1]
```diff
@@ -50,3 +73,24 @@ WITH (
...
+   connector='kafka',
+   topic='demo_topic',
+   properties.bootstrap.server='172.10.1.1:9090,172.10.1.2:9090',
+   scan.startup.mode='latest',
+   scan.startup.timestamp_millis='140000000',
+   properties.group.id='demo_consumer_name'
+)
+ROW FORMAT PROTOBUF MESSAGE 'main_message'
+ROW SCHEMA LOCATION 'file:///risingwave/proto-complex-schema';
+```
```
Comment (tabVersion) [inline docs/sql/create-source/create-source-kafka-redpanda.md]: suggest add a example loading from s3 for both avro and protobuf

**Review by tabVersion — APPROVED** [review]
rest LGTM

File: docs/sql/create-source/create-source-kafka-redpanda.md (hunk 1 above)
Comment (hengm3467) [reply docs/sql/create-source/create-source-kafka-redpanda.md]: @tabVersion  Shall we change the location from a local one to an S3 address?

Comment (tabVersion) [reply docs/sql/create-source/create-source-kafka-redpanda.md]: Although leaving as this version is good enough, we may add another example to illustrate we support loading from local and s3 for both avro and pb

Comment (hengm3467) [reply docs/sql/create-source/create-source-kafka-redpanda.md]: @tabVersion  Thanks for the clarification! We'll add a call-out note to tell users that they can use local locations as well. cc @bernscode for visibility.

**Pull request merged**